gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/AuditServiceImpl.java

```diff
@@ -93,14 +114,35 @@ public MetadataPage<AuditEntity> search(final ExecutionContext executionContext,
             criteria.environmentIds(Collections.singletonList(executionContext.getEnvironmentId()));
         }
 
-        if (query.isCurrentEnvironmentLogsOnly()) {
-            criteria.references(Audit.AuditReferenceType.ENVIRONMENT, Collections.singletonList(executionContext.getEnvironmentId()));
-        } else if (query.isCurrentOrganizationLogsOnly()) {
-            criteria.references(Audit.AuditReferenceType.ORGANIZATION, Collections.singletonList(executionContext.getOrganizationId()));
-        } else if (query.getApiIds() != null && !query.getApiIds().isEmpty()) {
-            criteria.references(Audit.AuditReferenceType.API, query.getApiIds());
-        } else if (query.getApplicationIds() != null && !query.getApplicationIds().isEmpty()) {
+        if (
+            permissionService.hasPermission(
+                executionContext,
+                RolePermission.ORGANIZATION_AUDIT,
+                executionContext.getOrganizationId(),
+                RolePermissionAction.READ
+            ) &&
+            query.getReferenceType() != null &&
+            query.getReferenceType().equals(AuditReferenceType.ORGANIZATION)
+        ) {
+            criteria.references(Audit.AuditReferenceType.ORGANIZATION, null);
```

Comment:
because the goal here is to have only ORGANIZATION audit event scooped by organizationId found in context. so the id of ogranization is not necessary. 
also allows to display the events that do not have referenceId only ORG type

Reply:
:heavy_check_mark: